doc/requirements/requirements.md

```diff
@@ -0,0 +1,306 @@
+* The user should be able to plan complex missions which include a series of movements, and the Navigation System should be able to accept those mission plans and execute them. 
+* The user should be able to specify different types of Robot drive types, such as Ackerman (automobile) steering, and Robot shapes (the current navigation stack is very limited in these areas).
+* The Navigation System should be able to handle more map types and orientations, including outdoor and 2D+ terrain.
+
+# 2.0 Requirements
+
+This section lists the requirements for the Navigation System.
+
+## 2.1 Implementation Constraints
+
+There are various constraints on the development of the ROS 2 Navigation stack.
+
+Id | Handle | Priority | Description | Notes
+-- | ------ | -------- | ----------- | -----
+IC001 | Developer's Guide | 1 | The Navigation System SHALL be developed in accordance with the ROS 2 Devloper's Guide | [ROS 2 Developer's Guide](https://github.com/ros2/ros2/wiki/Developer-Guide)
+IC002 | Implementation Language.C++.Version | 1 | Developers SHALL assume the availability of C++14 language features | Per the ROS 2 Developer's Guide
+IC003 | Implementation Language.C++.API Preference | 1 | Developers SHALL prefer standard C++, then Boost, then custom code, in that order. | Boost may be used if equivalent functionality is not already available in the C++ standard library
+IC004 | Implementation Language.C++.Supported Compilers.g++ | 1 | The Navigation System code SHALL compile with gcc 5.4 or newer
+IC004 | Implementation Language.C++.Supported Compilers.Clang | 1 | The Navigation System code SHALL compile with Clang, version *x*
+IC004 | Implementation Language.C++.Supported Compilers.Intel C++ Compiler | 1 | The Navigation System code SHOULD compile with the Intel C++ Compiler, version *x* | Could be useful for optimization purposes
+IC005 | Implementation Language.Python.Version | 1 | Any Python code developed for the Navigation System MUST use Python 3
+IC006 | Implementation Language.GUI | 1 | Any GUIs developed as part of the Navigation System MUST use the Qt library, via C++ or Python (PyQt) | *Which version?*
+IC006 | Implementation Language.GUI.QML | 1 | Any GUIs developed as part of the Navigation System MAY use QML
+
+## 2.2 Development Tools
+
+This section specifies the development tools and associated targets that will be used for the project.
+
+Id | Handle | Priority | Description | Notes
+-- | ------ | -------- | ----------- | -----
+DEV001 | Tools.ROS2.Version | 1 | The Navigation System WILL be developed against the latest stable version of the ROS 2 stack | *What is the current latest version?*
+DEV002 | Tools.Unit Test.Library | 1 | *TODO: GTest and GMock?*
```

Comment:
Implementation detail.
And [Catch 2](https://github.com/catchorg/Catch2) seems like an interesting alternative to Gtest

Reply:
Only if we're OK with using any unit test library (or various unit test libraries). If we care, we should call it out.